tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/ProduceDeliveryCallbacks.cs

```diff
@@ -0,0 +1,27 @@
+// <copyright file="ProduceDeliveryCallbacks.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using Datadog.Trace.Util.Delegates;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Kafka;
+
+internal struct ProduceDeliveryCallbacks : IBegin1Callbacks, IVoidReturnCallback
+{
+    public void OnException(object? sender, Exception ex)
+    {
+    }
+
+    public void OnDelegateEnd(object? sender, Exception? exception, object? state)
+    {
+    }
+
+    public object? OnDelegateBegin<TDeliveryReport>(object? sender, ref TDeliveryReport deliveryReport)
+    {
+        // no need to add logic here, we just need a dummy handler
```

Comment:
I'm a little confused, why do we need a handler proxy if we don't need it to execute 🤔

Reply:
If we have a handler (even an empty one) it will enable `KafkaProduceSyncDeliveryHandlerIntegration`. So in case DSM is enabled we'll have partition tags on Spans even if the user didn't set the delivery handler.
https://github.com/confluentinc/confluent-kafka-dotnet/blob/65362199f13bdad8a0831541f53d92e1e95a8a37/src/Confluent.Kafka/Producer.cs#L922C25-L922C25